packages/dd-trace/src/appsec/iast/taint-tracking/rewriter.js

```diff
@@ -175,29 +187,28 @@ function enableEsmRewriter (telemetryVerbosity) {
     port1.unref()
     port2.unref()
 
-    const chainSourceMap = isFlagPresent('--enable-source-maps')
-    const data = {
-      port: port2,
-      csiMethods,
-      telemetryVerbosity,
-      chainSourceMap
-    }
-
     try {
       Module.register('./rewriter-esm.mjs', {
```

Comment:
as this is not going to be specific thing for iast anymore, should we move this or at least import this from `initialize.mjs` or `loader-hooks.mjs`?

For iast we can assume that the rewriter is not working if node version is `<v20.6.0`, but can whole orchestrion assume that? Asking because `module.register` was added in 20.6.0.

Keep in mind that this code is not executed if there is no loader configured before, we added it for iast because there isn't any need to rewrite esm customer code when esm is not properly configured.

Reply:
I was talking about this with @rochdev and the biggest issue with doing this early on in the loading process is that we don't have config. We need to see if `config.iast.enabled` before initializing the rewriter. That might be a breaking change. We can probably figure out a way of initializing at least _that_ config early on, but changing how all that works is still a bigger change, and I'm not sure yet if it's breaking. What I've done instead is ensure that the existing langchain integration (i.e. by traditional RITM instrumentation) still works, so we're not reliant on Orchestrion for that. That way this PR is not a breaking change, and we can figure out how to enable Orchestrion for older Node.js at a later time, maybe in v6. How does that sound?

Reply:
> We need to see if config.iast.enabled before initializing the rewriter.

oh, I'm stupid, yes. It was also one of the reasons I wrote this here originally.

> Keep in mind that this code is not executed if there is no loader configured before, we added it for iast because there isn't any need to rewrite esm customer code when esm is not properly configured.

I emphasize on this, this code is not executed if customer has not `--loader` configured or `initialize.mjs` loaded. 
Is something that you can assume for now?

Reply:
> Is something that you can assume for now?

That's fine for now. We can adjust this later if needed, but for now we just assume this way of enabling ESM support was used.